src/egl/drivers/dri2/platform_android.c

```diff
@@ -651,6 +651,37 @@ droid_swap_buffers(_EGLDriver *drv, _EGLDisplay *disp, _EGLSurface *draw)
    return EGL_TRUE;
 }
 
+static EGLBoolean
+droid_set_damage_region(_EGLDriver *drv,
+                        _EGLDisplay *disp,
+                        _EGLSurface *draw, const EGLint* rects, EGLint n_rects)
+{
+   struct dri2_egl_display *dri2_dpy = dri2_egl_display(disp);
+   struct dri2_egl_surface *dri2_surf = dri2_egl_surface(draw);
+   android_native_rect_t* droid_rects = NULL;
+   if (n_rects != 0)
+      droid_rects = (android_native_rect_t *)
+         calloc(n_rects, sizeof(android_native_rect_t));
+
+   EGLint surfWidth = dri2_surf->base.Width;
+   EGLint surfHeight = dri2_surf->base.Height;
+   EGLint dIndex;
+
+   for (dIndex = 0; dIndex < n_rects; dIndex++) {
+      EGLint i = dIndex * 4;
+      droid_rects[dIndex].left = rects[i]; // left == x
+      droid_rects[dIndex].bottom = rects[i + 1]; // bottom == y
+      droid_rects[dIndex].right = rects[i] + rects[i + 2]; // left + width
+      droid_rects[dIndex].top = rects[i + 1] + rects[i + 3]; // bottom + height
+   }
+
+   native_window_set_surface_damage(dri2_surf->window, droid_rects, n_rects);
```

Comment:
If I understood the issue correctly, you will need to guard this as well. This is because older version of Android does not have this function so Mesa would not compile.